apps/files/src/components/LocationPicker/LocationPicker.vue

```diff
@@ -0,0 +1,376 @@
+            <translate
+              translate-context="Short column label in files table for the time at which a file was modified"
+              >Updated</translate
+            >
+          </sortable-column-header>
+        </div>
+      </template>
+      <template #rowColumns="{ item: rowItem, index }">
+        <div :ref="index === 0 ? 'firstRowNameColumn' : null" class="uk-width-expand">
+          <file-item
+            :key="rowItem.viewId"
+            :item="rowItem"
+            :has-two-rows="true"
+            :indicators="indicatorArray(rowItem)"
+            :are-indicators-clickable="false"
+            @click.native="selectFolder(rowItem)"
+          />
+        </div>
+        <div class="uk-text-meta uk-text-nowrap uk-width-small uk-text-right">
+          {{ rowItem.size | fileSize }}
+        </div>
+        <div class="uk-text-meta uk-text-nowrap uk-width-small uk-text-right">
+          {{ formDateFromNow(rowItem.mdate) }}
+        </div>
+      </template>
+      <template #noContentMessage>
+        <no-content-message icon="folder">
+          <template #message
+            ><span v-translate>There are no resources in this folder.</span></template
+          >
+        </no-content-message>
+      </template>
+    </file-list>
+  </div>
+</template>
+
+<script>
+import { mapMutations, mapState, mapActions, mapGetters } from 'vuex'
+import pathUtil from 'path'
+import { getResourceName } from '../../helpers/resourceInfo'
+import MixinsGeneral from '../../mixins'
+import MixinsFilesListIndicators from '../../mixins/filesListIndicators'
+import MoveSidebarMainContent from './MoveSidebarMainContent.vue'
+import FileList from '../FileList.vue'
+import FileItem from '../FileItem.vue'
+import SortableColumnHeader from '../FilesLists/SortableColumnHeader.vue'
+import NoContentMessage from '../NoContentMessage.vue'
+
+export default {
+  name: 'LocationPicker',
+
+  components: {
+    FileList,
+    FileItem,
+    SortableColumnHeader,
+    NoContentMessage
+  },
+
+  mixins: [MixinsGeneral, MixinsFilesListIndicators],
+
+  data: () => ({
+    originalLocation: ''
+  }),
+
+  computed: {
+    ...mapState('Files', [
+      'selectedResourcesForMove',
+      'locationPickerTargetFolder',
+      'currentFolder'
+    ]),
+    ...mapGetters('Files', [
+      'activeFiles',
+      'fileSortField',
+      'fileSortDirectionDesc',
+      'publicLinkPassword'
+    ]),
+    ...mapGetters(['user']),
+
+    currentAction() {
+      return this.$route.query.action
+    },
+
+    resources() {
+      const resources = JSON.parse(JSON.stringify(this.$route.query.resource))
+
+      // In case there is only one resource, ensure that the return will still be an array
+      if (typeof resources === 'string') {
+        return [resources]
+      }
+
+      return resources
+    },
+
+    resourcesCount() {
+      return this.resources.length
+    },
+
+    target() {
+      return JSON.parse(JSON.stringify(this.$route.query.target))
```

Comment:
another weird copy

Reply:
See comment above

Reply:
should we maybe create our own utility function then ? the name of the function will already be explicit enough so we know what and why this is used

Reply:
Done